src/main/java/org/schemaspy/output/dot/schemaspy/DotNode.java

```diff
@@ -75,6 +75,7 @@ public DotNode(Table table, boolean fromRoot, DotNodeConfig config, DotConfig do
         this.dotConfig = dotConfig;
         this.path = createPath(fromRoot);
         this.columnSpan = config.showColumnDetails ? "COLSPAN=\"2\" " : "COLSPAN=\"3\" ";
+        this.excludedColumns = table.getColumns().stream().filter(TableColumn::isExcluded).collect(Collectors.toSet());
```

Comment:
Avoid operations other than assignment in the (primary) constructor. I'm fine if you do this in a secondary constructor, but if `excludeColumn` can be removed (see other comment) please also consider moving this to its own method.

Reply:
I was trying to limit the scope.
But it might not be that big of a creep from removing excluded columns from WriteStats.
DotNode has this usage of excludedColumns:
```
if (excludedColumns.contains(column))
```

Which could be changed to
```
if(column.isExcluded())
```
Which would remove `excludedColumns` completely.

Reply:
But for the constructor "assignment".
I was initially stuck between a few different solutions. Having it as constructor arg, method in Table or going for a more media/medium approach and using the exclude Column but as withExcludedColumns.

But decided that keeping it as little as possible would leave better room for the usage analysis.

Reply:
Testability starts to suffer quickly when we add logic to the primary constructor, so I would like to not see that in this PR.

Regarding which approach we want to transition to, I agree that that would be another PR

Reply:
I totally agree constructors should be simple.

I've created a cleanup issue to analys and fix DotNod in regards to excludedColumns #884.

If the scope is retained in #884 hopefully more issues are created since one could argue that DotNode is strange in both representing Table and Columns.

Which in it self makes the testing harder.